mtj/main.py

```diff
@@ -32,22 +32,35 @@ def main() -> None:
     a_par = 1e-3  # A_parallel(V), in (T)
     a_perp = 2e-4  # A_perpendicular(V), in (T)
 
+    # dummy values for Heff
+    K_u = 0 # crystal anisotropy constant
+    M_s = 1 # saturation magnetization module (A/m)
+    u_k = np.array([0, 0, 1]) # easy axis
+    Volt = 0 # STT voltage applied (V)
+    H_app = np.array([0, 0, 0]) # applied field (A/m)
+    N = np.array([[1,0,0],[0,1,0],[0,0,1]]) # demagnetization tensor
+
     for i, t in enumerate(time_series[:-1]):
         H_th = compute_thermal_field(alpha, T, Ms, V, dt)
         print("Thermal Field Vector H_th:", H_th)
         # Calculate the effective field
-        H_eff = calc_Heff(np.array([1, 0, 0]))
+        H_eff = calc_Heff(m[i],
+                          M_s,
+                          u_k,
+                          p,
+                          a_par,
+                          a_perp,
+                          Volt,
+                          H_app,
+                          N)
```

Comment:
It seems like you forgot to include `K_u` in the function call.
```suggestion
        H_eff = calc_Heff(m[i],
                          K_u,
                          M_s,
                          u_k,
                          p,
                          a_par,
                          a_perp,
                          Volt,
                          H_app,
                          N)
```